pkg/router/template/router.go

```diff
@@ -512,12 +512,27 @@ func (r *templateRouter) writeConfig() error {
 		r.state[k] = cfg
 	}
 
+	data, err := ioutil.ReadFile("/proc/sys/net/ipv4/neigh/default/gc_thresh3")
+	if err != nil {
+		glog.Warningf("Error reading ARP neighbour information: %s", err)
+	}
+	arpcache, err := strconv.Atoi(string(data[:len(data)-1]))
+	if err != nil {
+		glog.Warningf("Error: %s", err)
+	}
+	arpthreshold := float64(arpcache) * 0.9
+
 	for path, template := range r.templates {
 		file, err := os.Create(path)
 		if err != nil {
 			return fmt.Errorf("error creating config file %s: %v", path, err)
 		}
 
+		items := len(r.serviceUnits)
```

Comment:
I'm not convinced this is the right check.  Don't you have to count all of the endpoints under r.serviceUnits (see template_helper.go endpointsForAlias() )?

Anyway, if there are three routes that all point to the same pod on the backend, that will only cost you 1 entry in the arp cache.  We may have to loop over all of the endpoints, get the IP addresses, add them to a map, and then count the entries in the map to get the right count.

And that sounds somewhat expensive... We may need to do it under a rate limit, but can discuss that later.

Reply:
That's a lot of work for a warning.  At some point, why aren't we just alerting on this condition and calculating it at a node level?